manifests/agent.pp

```diff
@@ -244,37 +244,28 @@
   }
 
   ## ossec.conf generation concats
-  case $::kernel {
-    'Linux': {
-      case $::osfamily {
-        'Redhat', 'redhat', 'OracleLinux': {
-          $apply_template_os = 'rhel'
-          if ( $::operatingsystemrelease =~ /^7.*/ ) {
-            $rhel_version = '7'
-          } elsif ( $::operatingsystemrelease =~ /^6.*/ ) {
-            $rhel_version = '6'
-          } elsif ( $::operatingsystemrelease =~ /^5.*/ ) {
-            $rhel_version = '5'
-          } else {
-            fail('This ossec module has not been tested on your distribution')
-          }
-        }
-        'Debian', 'debian', 'Ubuntu', 'ubuntu': {
-          $apply_template_os = 'debian'
-          if ( $::lsbdistcodename == 'wheezy') or ($::lsbdistcodename == 'jessie') {
-            $debian_additional_templates = 'yes'
-          }
-        }
-        'Amazon': {
-          $apply_template_os = 'amazon'
-        }
-        'CentOS', 'Centos', 'centos': {
-          $apply_template_os = 'centos'
-        }
-        default: { fail('This ossec module has not been tested on your distribution') }
+  case $::operatingsystem{
+    'Redhat', 'redhat', 'OracleLinux':{
```

Comment:
Please check if `redhat` does actually exist in Puppet as a possible`::operativesystem`. I suspect it may not be neccessary anymore. If doesn't exist please remove it in the Manager also

Reply:
Hi @jm404 !

After starting a RHEL7 virtual machine and install Puppet on it, when checking Puppet facts we get:

```bash
[root@rhel ~]# puppet facts find operatingsystem
{"name":"operatingsystem","values":{"uptime_seconds":"5530",
"architecture":"x86_64",
"virtual":"kvm","is_virtual":"true","aug368",
"blockdevice_sda_vendor":"ATA",
"blockdevice_sda_model":"VBOX HARDDISK",
"blockdevices":"sda",

"dhcp_servers":"{
\"system\"eratingsystem":"RedHat",

"os":"{\"name\"=>\"RedHat\", 
\"family\"=>\"RedHat\",

 \"release\"=>{\"major\"=>\"7\", .... 

Corporation","boardproductname":"VirtualBox","boardserialnumbe[root@rhel ~]#
```

As you can see the operating system is called RedHat, so the case check:
```ruby
case $::operatingsystem{
    'Redhat', 'redhat', 'OracleLinux':{
```

It's correct and necessary.

kr,

Rshad

Reply:
The `puppet facts` output seems to be truncated and bad formatted. Please paste the `::operatingsystem` section without truncating it.

I assume that `"dhcp_servers":"{
\"system\"eratingsystem":"RedHat",` refers to the operating system, which in that case, would be  `RedHat`. 

So in our manifests we have: 
```
case $::operatingsystem{
    'Redhat', 'redhat', 'OracleLinux':{
```

So, neither `Redhat` nor `redhat` will match it, please review it and test the installation on a RedHat host after fixing it.

Best regards,

Jose

Reply:
Hi @jm404 !

After the changes made in https://github.com/wazuh/wazuh-puppet/pull/197/commits/cfed5f35f356a7f02cfa60dc58467d9f3a880362, I tested the instsallation of a wazuh-manager and a wazuh-agent in RHEL operating system and both installations were done successfully.

**My test Operating System characteristics:**

```
[root@rhel ~]# cat /etc/os-release
NAME="Red Hat Enterprise Linux Server"
VERSION="7.7 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VARIANT="Server"
VARIANT_ID="server"
VERSION_ID="7.7"
PRETTY_NAME="Red Hat Enterprise Linux Server 7.7 (Maipo)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.7:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.7
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.7"
[root@rhel ~]#

```

**Puppet Facts (puppet 6.11.1)**
```
{
  architecture => "x86_64",
  family => "RedHat",
  hardware => "x86_64",
  name => "RedHat",
  release => {
    full => "7.7",
    major => "7",
    minor => "7"
  },
  selinux => {
    config_mode => "enforcing",
    config_policy => "targeted",
    current_mode => "enforcing",
    enabled => true,
    enforced => true,
    policy_version => "31"
  }
}
```

Kr,

Rshad